bpf: Retire CT entries quickly for unreplied connections

joestringer opened the pull request:
Set a five minute timeout for conntrack entries that have not yet seen a
non-SYN packet. For non-TCP protocols, the timeouts remain longer. ICMP
response tuples for TCP connections will be timed out quickly in all cases.

Enable active garbage collection of CT LRU maps from userspace. 

Fixes: #3232
Related: #3236

joestringer committed:
bpf: Retire entries quickly for unreplied connections

Set a five minute timeout for conntrack entries that have not yet seen a
non-SYN packet. For non-TCP protocols, the timeouts remain longer. ICMP
response tuples for TCP connections will be timed out quickly in all cases.

Fixes: #3232

Signed-off-by: Joe Stringer <joe@covalent.io>

joestringer committed:
bpf: Remove dead code

Signed-off-by: Joe Stringer <joe@covalent.io>

tgraf commented on bpf/lib/conntrack.h:
```diff
@@ -51,9 +52,13 @@ static inline void __inline__ __ct_update_timeout(struct ct_entry *entry,
+	if (entry->seen_non_syn)
+		__ct_update_timeout(entry, CT_DEFAULT_LIFETIME);
+	else
+		__ct_update_timeout(entry, CT_SYN_TIMEOUT);
```
`__ct_update_timeout()` is protected by `NEEDS_TIMEOUT` which is undefined if we are running with LRU. The expiration needs to run with LRU as well.

joestringer replied:
Good point, I added another commit to enable these timeouts in LRU case and to enable garbage collection in userspace for conntrack maps backed by LRU map type.

joestringer committed:
bpf: Enable GC for LRU conntrack maps

Until github issue #3239 is resolved, set timeouts in conntrack entries
and enable active garbage collection in userspace.

Signed-off-by: Joe Stringer <joe@covalent.io>
